### PR TITLE
[ROX-17704] Remove dackbox indexers and searchers

### PR DESCRIPTION
**Pull request by md2119**
## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### CODE REVIEW

**Comment by connorgorman** [comment]
needs a rebase

File: central/deployment/datastore/datastore_bench_test.go [hunk 1]
```diff
@@ -26,6 +27,8 @@ import (
 )
 
 func BenchmarkSearchAllDeployments(b *testing.B) {
+	pgtest.SkipIfPostgresEnabled(b)
+
 	ctx := sac.WithGlobalAccessScopeChecker(context.Background(),
```
Comment (connorgorman) [inline central/deployment/datastore/datastore_bench_test.go:32]: Either we need to migrate this or just delete it

File: central/deployment/datastore/internal/search/searcher.go [hunk 2]
```diff
@@ -35,26 +35,14 @@ type Searcher interface {
...
-		searcher: formatSearcher(graphProvider,
-			cveIndexer,
-			componentCVEEdgeIndexer,
-			componentIndexer,
-			imageComponentEdgeIndexer,
-			imageIndexer,
-			deploymentIndexer,
-			imageCVEEdgeIndexer),
-	}
+func New(_ store.Store,
```
Comment (connorgorman) [inline central/deployment/datastore/internal/search/searcher.go:38]: if we return nil, we should probably just delete

File: central/deployment/index/search_comparison_test.go [hunk 3]
```diff
@@ -83,6 +79,8 @@ func TestImageSearchResults(t *testing.T) {
 }
 
 func TestDeploymentSearchResults(t *testing.T) {
+	pgtest.SkipIfPostgresEnabled(t)
```
Comment (connorgorman) [inline central/deployment/index/search_comparison_test.go:82]: delete or update

File: central/cve/datastore/singleton.go [hunk 4]
```diff
@@ -28,24 +15,10 @@ var (
 
 func initialize() {
 	globalDackbox := globaldb.GetGlobalDackBox()
```
Comment (dashrews78) [inline central/cve/datastore/singleton.go:17]: Why do we still need dackbox here?  Is this one of those CVE things we do differently now that needs to be deprecated itself?

File: central/deployment/datastore/datastore_bench_test.go (hunk 1 above)
Comment (md2119) [reply central/deployment/datastore/datastore_bench_test.go:32]: There is TODO JIRA for it

File: central/deployment/datastore/internal/search/searcher.go (hunk 2 above)
Comment (md2119) [reply central/deployment/datastore/internal/search/searcher.go:38]: one step at a time :)

File: central/deployment/index/search_comparison_test.go (hunk 3 above)
Comment (md2119) [reply central/deployment/index/search_comparison_test.go:82]: There is TODO JIRA

File: central/cve/datastore/singleton.go (hunk 4 above)
Comment (md2119) [reply central/cve/datastore/singleton.go:17]: This PR does not remove stores, only indexers and searchers :)

File: central/node/datastore/datastore.go [hunk 5]
```diff
@@ -50,19 +46,11 @@ type DataStore interface {
...
-	searcher := search.New(dataStore,
-		dacky,
-		cveIndexer.New(bleveIndex),
-		componentCVEEdgeIndexer.New(bleveIndex),
-		componentIndexer.New(bleveIndex),
-		nodeComponentEdgeIndexer.New(bleveIndex),
-		indexer,
-	)
-	ds := newDatastoreImpl(dataStore, indexer, searcher, risks, nodeRanker, nodeComponentRanker)
+	var searcher search.Searcher
```
[the remark this answers is not in the document]
Comment (md2119) [reply central/node/datastore/datastore.go:52]: The entire func will go away :)

File: central/deployment/datastore/datastore_bench_test.go (hunk 1 above)
Comment (connorgorman) [reply central/deployment/datastore/datastore_bench_test.go:32]: Can you put a todo comment in the code then like you did for the others?

File: central/deployment/index/search_comparison_test.go (hunk 3 above)
Comment (connorgorman) [reply central/deployment/index/search_comparison_test.go:82]: also add to the code the ticket

**Review by connorgorman — APPROVED** [review]
LGTM, please add the jira tickets one the two benchmarks

File: central/cve/datastore/singleton.go (hunk 4 above)
Comment (dashrews78) [reply central/cve/datastore/singleton.go:17]: one step at a time.  got it.

**Review by dashrews78 — APPROVED** [review]
LGTM

**Pull request merged**